src/features/auth/authSlice.ts

```diff
@@ -63,22 +75,50 @@ export const useTryAuthFromToken = (token?: string) => {
     skip: !normToken,
   });
 
-  if (tokenQuery.isSuccess && normToken !== currentToken) {
-    dispatch(
-      setAuth({
-        token: normToken,
-        username: tokenQuery.data.user,
-        tokenInfo: tokenQuery.data,
-      })
-    );
-  }
+  useEffect(() => {
+    if (tokenQuery.isSuccess && normToken !== currentToken) {
+      dispatch(
+        setAuth({
+          token: normToken,
+          username: tokenQuery.data.user,
+          tokenInfo: tokenQuery.data,
+        })
+      );
+    }
+  }, [
+    currentToken,
+    dispatch,
+    normToken,
+    tokenQuery.data,
+    tokenQuery.isSuccess,
+  ]);
 
   return tokenQuery;
 };
 
+export const useSetTokenCookie = () => {
+  const token = useAppSelector(authToken);
+  const expires = useAppSelector(({ auth }) => auth.tokenInfo?.expires);
+  useEffect(() => {
+    if (token && expires) {
+      setCookie('kbase_session', token, {
+        expires: new Date(expires),
+        ...(process.env.NODE_ENV === 'development'
+          ? {}
+          : { domain: process.env.REACT_APP_KBASE_DOMAIN }),
+      });
+    } else if (!token) {
+      clearCookie('kbase_session');
+    } else {
+      // eslint-disable-next-line no-console
+      console.error('Could not set token cookie, missing expire time');
+    }
+  }, [expires, token]);
+};
+
 const normalizeToken = <T = undefined>(
-  t?: string,
-  fallback: T = undefined as T
+  t: string | undefined,
+  fallback?: T
 ): string | T => {
-  return t?.toUpperCase().trim() || fallback;
+  return t?.toUpperCase().trim() || (fallback as T);
 };
```

Comment:
I want to clean this up before merging, its too complicated for what it is (and the type assertion in gross)